src/asr_got_reimagined/domain/stages/stage_5_pruning_merging.py

```
@@ -23,309 +23,206 @@ class PruningMergingStage(BaseStage):
+               n.confidence_theoretical_basis AS conf_theo,
+               n.confidence_methodological_rigor AS conf_meth,
+               n.confidence_consensus_alignment AS conf_cons,
+               n.metadata_impact_score AS impact_score
         """
-        Determines if a node should be pruned based on P1.5.
-        Pruning threshold: min(E[C]) < threshold & low impact (P1.28).
-        Note: P1.5 says "min(E[C])", which means the minimum of the *expected values* of
-        the confidence components if they were distributions.
-        For our current ConfidenceVector (scalar components), we can interpret E[C] as C itself.
-        So, min(C_components) < threshold.
+        nodes_to_prune_ids: List[str] = []
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)